brightwind/transform/transform.py

```diff
@@ -34,6 +34,28 @@ def _compute_wind_vector(wspd, wdir):
     return wspd*np.cos(wdir), wspd*np.sin(wdir)
 
 
+def dataframe_map(df, func, **kwargs):
+    """
+    Apply a function element-wise to a DataFrame.
+    
+    Compatibility wrapper for DataFrame.map() (pandas >=2.1) and 
+    DataFrame.applymap() (pandas <2.1).
+
+    :param df:              DataFrame to apply function to
+    :type df:               pd.DataFrame
+    :param func:            Function to apply element-wise
+    :type func:             Callable
+    :return:                Transformed DataFrame
+    :rtype:                 pd.DataFrame
+    """
+    if hasattr(df, 'map'):
+        # pandas >= 2.1
+        return df.map(func, **kwargs)
+    else:
+        # pandas < 2.1
+        return df.applymap(func, **kwargs)
```

Comment:
@olivia-bentley could we add a depreciation warning here like I did for the freq strings to let users know that they need to update?